asv_bench/benchmarks/rolling.py

```diff
@@ -150,19 +150,23 @@ def time_quantile(self, constructor, window, dtype, percentile, interpolation):
         self.roll.quantile(percentile, interpolation=interpolation)
 
 
-class PeakMemFixed:
-    def setup(self):
-        N = 10
-        arr = 100 * np.random.random(N)
-        self.roll = pd.Series(arr).rolling(10)
-
-    def peakmem_fixed(self):
-        # GH 25926
-        # This is to detect memory leaks in rolling operations.
-        # To save time this is only ran on one method.
-        # 6000 iterations is enough for most types of leaks to be detected
-        for x in range(6000):
-            self.roll.max()
+class PeakMemFixedWindowMinMax:
+
+    params = [
+        pd.core.window.rolling.Rolling.min,
+        pd.core.window.rolling.Rolling.max,
+    ]
+
+    def setup(self, f):
+        N = int(1e6)
+        arr = np.random.random(N)
+        self.roll = pd.Series(arr).rolling(2)
+
+    def peakmem_fixed(self, f):
+        # GH 33693
+        # increased size of array so we see a clearer picture
+        for x in range(5):
```

Comment:
i don’t think u need this extra
look at all

Reply:
Sorry, I don't really get what you mean. I removed the comments, is that right?

Reply:
remove L165

Reply:
So that everything is in the setup function?

Reply:
you don’t need this just remove it
and show the benchmark runs

Reply:
Does "this" refer to line 165 or the full benchmark?

If I remove line 165 there would be no benchmark anymore, only the setup function, and asv doesn't run anything anymore.

Reply:
remove the 'for x in range(5)' the asv is sufficient as is w/o an extra loop.

Reply:
Without the for loop (not committed yet), I get these results:
```
 $ asv show master                                                    
Commit: master <master>

rolling.PeakMemFixedWindowMinMax.peakmem_fixed [my_machine/virtualenv-py3.8-Cython-matplotlib-numba-numexpr-numpy-odfpy-openpyxl-pytest-scipy-sqlalchemy-tables-xlrd-xlsxwriter-xlwt]
  ok
  ======== ======
   param1        
  -------- ------
    min     198M 
    max     197M 
  ======== ======
  started: 2020-04-28 16:03:03, duration: 418ms


 $ asv show rolling-min-max-memory-leak                               
Commit: rolling- <rolling-min-max-memory-leak>

rolling.PeakMemFixedWindowMinMax.peakmem_fixed [my_machine/virtualenv-py3.8-Cython-matplotlib-numba-numexpr-numpy-odfpy-openpyxl-pytest-scipy-sqlalchemy-tables-xlrd-xlsxwriter-xlwt]
  ok
  ======== ======
   param1        
  -------- ------
    min     197M 
    max     198M 
  ======== ======
  started: 2020-04-28 16:04:34, duration: 419ms
```

There's no difference anymore, is this intended?

Reply:
ok then i guess leave it